terra/config.py

```diff
@@ -18,39 +18,50 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 """
-from gi.repository import Gdk
+
 from base64 import b64encode, b64decode
 import ConfigParser
-from defaults import ConfigDefaults
 import os
 
-__terra_data_directory__ = '/usr/local/share/terra/'
-__terra_app_directory__ = '/usr/local/share/applications/'
-__version__ = '0.2.0'
+from gi.repository import Gdk
 
-class ConfigManager():
-    config_dir_path = os.path.join(os.environ['HOME'], '.config', 'terra')
-    config_file_path = os.path.join(config_dir_path, 'main.cfg')
+from terra import (__version__)
+from terra.defaults import ConfigDefaults
+from terra.handler import TerraHandler
+
+
+class ConfigManager:
+    version = __version__
+    config_file_name = 'main.cfg'
 
-    config = ConfigDefaults
-    config.read(config_file_path)
     defaults = ConfigDefaults
+    config = ConfigDefaults
 
     callback_list = []
     use_fake_transparency = False
     disable_losefocus_temporary = False
 
-    data_dir = __terra_data_directory__
-    version = __version__
+    @classmethod
+    def __init__(cls):
+        # Load the user config, if any.
+        config_path = TerraHandler.get_config_path()
+        if os.path.exists(config_path):
+            config_file_path = os.path.join(config_path, cls.config_file_name)
+            if os.path.exists(config_file_path):
+                cls.config.read(config_file_path)
 
-    @staticmethod
-    def init():
-        ConfigParser.SafeConfigParser({})
-        ConfigManager.save_config()
+        # TODO: Remove once the code no longer requires the config files!
```

Comment:
There should be no need for a user config file (`~/.config/terra/main.cfg`) when you just initialize the ConfigManager, because you don't make any changes, the application should run just fine without that file. It should be created only when you change the settings.